vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -1541,20 +1541,26 @@ void vtkF3DRenderer::Render()
 
   auto cpuElapsed = std::chrono::high_resolution_clock::now() - cpuStart;
 
-  // Get CPU frame per seconds
-  int fps = static_cast<int>(std::round(
-    1.0 / (std::chrono::duration_cast<std::chrono::microseconds>(cpuElapsed).count() * 1e-6)));
+  vtkInformation *info = this->GetInformation();
+
+  bool uiOnly = info->Get(vtkF3DRenderPass::RENDER_UI_ONLY());
+
+  if (! uiOnly)
+  {
+	  // Get CPU frame time
+	  double elapsedTime = std::chrono::duration_cast<std::chrono::microseconds>(cpuElapsed).count() * 1e-6;
 
 #if !defined(__ANDROID__) && !defined(__EMSCRIPTEN__)
-  glEndQuery(GL_TIME_ELAPSED);
-  GLint elapsed;
-  glGetQueryObjectiv(this->Timer, GL_QUERY_RESULT, &elapsed);
+	  glEndQuery(GL_TIME_ELAPSED);
+	  GLint elapsed;
+	  glGetQueryObjectiv(this->Timer, GL_QUERY_RESULT, &elapsed);
 
-  // Get min between CPU frame per seconds and GPU frame per seconds
-  fps = std::min(fps, static_cast<int>(std::round(1.0 / (elapsed * 1e-9))));
+	  // Get min between CPU frame time and GPU frame time
+	  elapsedTime = std::min(elapsedTime, elapsed * 1e-9);
 #endif
 
-  this->UIActor->SetFpsValue(fps);
+	  this->UIActor->UpdateFpsValue(elapsedTime);
```

Comment:
would it make sense to only compute the FPS when we need it (aka when the fps counter is displayed) ?

Reply:
It's cheap, I think it's fine to update it in the background so we can have a good value directly when the user make it visible.